editors/code/src/main.ts

```diff
@@ -200,3 +191,34 @@ function createCommands(): Record<string, CommandFactory> {
         revealDependency: { enabled: commands.revealDependency },
     };
 }
+
+function conflictExtDetect() {
+    if (vscode.extensions.getExtension("rust-lang.rust")) {
+        vscode.window
+            .showWarningMessage(
+                `You have both the rust-analyzer (rust-lang.rust-analyzer) and Rust (rust-lang.rust) ` +
+                    "plugins enabled. These are known to conflict and cause various functions of " +
+                    "both plugins to not work correctly. You should disable one of them.",
+                "Got it",
+            )
+            .then(() => {}, console.error);
+    }
+
+    if (vscode.extensions.getExtension("panicbit.cargo")) {
+        const isRustAnalyzerCheckOnSave = vscode.workspace
+            .getConfiguration("rust-analyzer")
+            .get("checkOnSave");
+        const isCargoAutomaticCheck = vscode.workspace
+            .getConfiguration("cargo")
+            .get("automaticCheck");
+        if (isRustAnalyzerCheckOnSave && isCargoAutomaticCheck) {
```

Comment:
maybe a dumb question: is it possible to _just_ check if the Cargo extension is installed? I assume that most of the functionality is now in rust-analyzer, modulo automatically adding dependencies.

Reply:
done

Reply:
Feels like this might be really annoying for someone who wants to use the features of the other extension.